src/sentry/static/sentry/app/views/settings/projectAlerts/settings.tsx

```diff
@@ -12,34 +12,39 @@ import Form from 'app/views/settings/components/forms/form';
 import JsonForm from 'app/views/settings/components/forms/jsonForm';
 import PermissionAlert from 'app/views/settings/project/permissionAlert';
 import PluginList from 'app/components/pluginList';
-import SentryTypes from 'app/sentryTypes';
 import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
 import routeTitleGen from 'app/utils/routeTitle';
+import {Organization, Project, Plugin} from 'app/types';
 
-class ProjectAlertSettings extends AsyncView {
-  static propTypes = {
-    ...AsyncView.propTypes,
-    // these are not declared as required of issues with cloned elements
-    // not initially defining them (though they are bound before) ever
-    // rendered
-    organization: SentryTypes.Organization,
-    project: SentryTypes.Project,
-    canEditRule: PropTypes.bool.isRequired,
-  };
+type RouteParams = {orgId: string; projectId: string};
+type Props = RouteComponentProps<RouteParams, {}> & {
+  canEditRule: boolean;
+  organization: Organization;
+  project: Project;
+} & AsyncView['props'];
+
+type State = {
+  project: Project;
+  pluginList: Array<Plugin>;
```

Comment:
I think all data that is loaded via async component has to have `| null` as they are null on error.

Reply:
good catch @scttcper :) Thanks!  `AsyncComponent` also set them to null in the `getDefaultState`

```
endpoints.forEach(([stateKey, _endpoint]) => {
      state[stateKey] = null;
    });
```

Reply:
Although you are right, I am wondering if this is correct in the `AsyncComponent`, because after checking some of the components that use `AsyncView` | `AsycComponent` I noticed that they don't use 'null' + if we use null then we would always have to check the state properties and that is very verbose. Let me bring this topic to the TSC meeting 😉

Reply:
good idea, I have been bitten by this before 👍